handler/src/main/java/io/netty/handler/ssl/SniHandler.java

```diff
@@ -100,7 +100,11 @@ public SslContext sslContext() {
     protected final void onLookupComplete(ChannelHandlerContext ctx,
                                           String hostname, Future<SslContext> future) throws Exception {
         if (!future.isSuccess()) {
-            throw new DecoderException("failed to get the SslContext for " + hostname, future.cause());
+            if (future.cause() instanceof Error) {
```

Comment:
@jasontedor nit: Can you store the cause in a temp variable as we access it multiple times and each time this is a volatile access.

Reply:
I've made this change, squashed, and force pushed.